zilencer/views.py

```diff
@@ -166,10 +183,24 @@ def remote_server_notify_push(request: HttpRequest, entity: Union[UserProfile, R
         server=server,
     ))
 
+    if encrypted:
+        android_data = prepare_encrypted_data(android_devices, gcm_payload)
+        apple_data = prepare_encrypted_data(apple_devices, apns_payload)
+
+        if android_data:
+            send_android_encrypted_push_notification(android_data, gcm_options)
+
+        if apple_data:
+            send_apple_encrypted_push_notification(user_id, apple_data)
+
+        return json_success()
+
     if android_devices:
+        android_devices = [d for d in android_devices if not d.encrypted]
         send_android_push_notification(android_devices, gcm_payload, gcm_options, remote=True)
 
     if apple_devices:
+        apple_devices = [d for d in apple_devices if not d.encrypted]
         send_apple_push_notification(user_id, apple_devices, apns_payload, remote=True)
```

Comment:
If there are some devices but they're all encrypted, then this defeats the purpose of the `if`.